### PR TITLE
Use hardcoded string as username for Azure git credentials

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Since Azure git provider do not support `username` for its user object, use `username` string as username in the git credentials file if the PAT is related to Azure Devops:
```
https://username:<token>@dev.azure.com
```

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/22313

### How to test this PR?

1. Setup an Azure DevOps PAT secret.
2. Start a workspace from a private Azure DevOps repo.
See: the workspace starts and the project is git cloned successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java [hunk 1]
```diff
@@ -158,7 +159,9 @@ public void createOrReplace(PersonalAccessToken personalAccessToken)
                               scmUrl.getProtocol(),
                               personalAccessToken.getScmTokenName().startsWith(OAUTH_2_PREFIX)
                                   ? "oauth2"
-                                  : personalAccessToken.getScmUserName(),
+                                  : isNullOrEmpty(personalAccessToken.getScmOrganization())
+                                      ? personalAccessToken.getScmUserName()
+                                      : personalAccessToken.getScmOrganization(),
```
Comment (amisevsk) [inline infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java:164]: I haven't tested this yet but this appears to apply to all platforms (e.g. GitHub, etc.). Is this intentional? I'm generally a little confused on what works for all platforms -- e.g. on GitHub my username is `amisevsk` and (I guess) my organization name is `amisevsk`, even though I'm not an organization?

Comment (vinokurig) [reply infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java:164]: According to the [PAT documentation](https://www.eclipse.org/che/docs/stable/end-user-guide/using-a-git-provider-access-token/) we use the `che.eclipse.org/scm-organization` annotation only for the Azure DevOps. For GitHub and others except Azure, `scmUsername` is used so the flow is the same.

Comment (ibuziuk) [reply infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java:164]: can we extract this logic to a separate method and add javadoc for that?

Comment (vinokurig) [reply infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesGitCredentialManager.java]: done

**Comment by vinokurig** [comment]
Updated the PR title and description according to the logic changes.

**Review by tolusha — APPROVED** [review]

**Review by amisevsk — APPROVED** [review]
LGTM; haven't had a chance to test directly though.

**Comment by ibuziuk** [comment]
@vinokurig could you please provide the image for testing? My understanding that with this PR token should work when configured from the user dashboard, right?

**Comment by vinokurig** [comment]
A docker image to test the PR is `quay.io/ivinokur/che-server:che-22313`

**Pull request merged**